### PR TITLE
Restrict documents that can be uploaded

### PR DESCRIPTION
**Pull request by cpjmcquillan**
Restrict content-types accepted by the file input and implement server side check on content-type.

Error message has been signed off by Content.

## Jira ticket URL
[TEVA-697](https://dfedigital.atlassian.net/browse/TEVA-697)

## Changes in this PR:
- Accept only .doc, .xls, .ppt, .pdf, .jpeg and .mp4 file types in the file input so users can't select other file types
- Restrict this server side as well in case an invalid file type was able to be submitted

## Screenshots of UI changes:
image

### CODE REVIEW

File: app/views/hiring_staff/vacancies/documents/show.html.haml [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
       = f.govuk_file_field :documents,
         label: { text: t('jobs.upload_file'), size: 's' },
+        accept: '.doc, .xls, .ppt, .pdf, image/jpeg, image/png, video/mp4',
```
Comment (tatyree) [inline app/views/hiring_staff/vacancies/documents/show.html.haml:16]: Shouldn't this have `.docx` and `.xlsx` as well (is there a `.pptx`?).

File: config/locales/en.yml [hunk 2]
```diff
@@ -265,6 +265,7 @@ en:
     file_google_error_message: '%{filename} could not be uploaded - try again'
     file_input_error_message: 'The selected file(s) could not be uploaded'
     file_size_error_message: '%{filename} must be smaller than %{size_limit}'
+    file_type_error_message: '%{filename} must be a valid file type'
```
Comment (david-mears-dfe) [inline config/locales/en.yml:268]: Could we list the valid filetypes here?

File: app/views/hiring_staff/vacancies/documents/show.html.haml (hunk 1 above)
Comment (cpjmcquillan) [reply app/views/hiring_staff/vacancies/documents/show.html.haml:16]: There is, and I asked the question but no response. Will add now 👍

File: config/locales/en.yml (hunk 2 above)
Comment (cpjmcquillan) [reply config/locales/en.yml:268]: It could be a long list...

Comment (cpjmcquillan) [reply config/locales/en.yml:268]: Also, should add that an end user would probably never see this message.

**Pull request merged**